src/AppLovinMAXAdView.js

```diff
@@ -111,6 +111,19 @@ const AdView = (props) => {
     if (props.onAdRevenuePaid) props.onAdRevenuePaid(event.nativeEvent);
   };
 
+  const sanitizeExtraParameters = (name, params) => {
+    if (params) {
+      for (const key in params) {
+        const value = params[key];
+        if ((value != null) && (value != undefined) && (typeof value !== 'string')) {
```

Comment:
Same comment as below.

Reply:
Checking `null` and `undefined` is necessary since we allow `null` for setXXXLocalExtraParameter().   But, since we've already removed runIfInitialized() and it is nice to have logging messages on the platform side when it comes to customer support, can we check this on the platform side?

```
RCT_EXPORT_METHOD(setBannerLocalExtraParameter:(NSString *)adUnitIdentifier :(NSString *)key :(nullable id)value)
{
    NSString *stringValue = nil;

    if (value)
    {
        if ([value isKindOfClass:[NSString class]]) {
            stringValue = [RCTConvert NSString:value];
        }

        if (!stringValue)
        {
            [self log: @"setBannerLocalExtraParameter supports only string values for \"%@\"", key];
            return;
        }
    }

    [self setAdViewLocalExtraParameterForAdUnitIdentifier: adUnitIdentifier adFormat: DEVICE_SPECIFIC_ADVIEW_AD_FORMAT key: key value: stringValue];
}
```

This will allow string, null, and undefined in JavaScript.  And the error messages are:

```
[AppLovinSdk] [AppLovinMAX] setBannerLocalExtraParameter supports only string values for "banner2"
[AppLovinSdk] [AppLovinMAX] setBannerLocalExtraParameter supports only string values for "banner3"
[AppLovinSdk] [AppLovinMAX] setBannerLocalExtraParameter supports only string values for "banner4"
[AppLovinSdk] [AppLovinMAX] setBannerLocalExtraParameter supports only string values for "banner7"
```

Reply:
This won't work on Android.   React Native converts variables before passing to `setBannerLocalExtraParameter`.

Reply:
ok, two things:

- In our codebase, we usually practice [early returns](https://medium.com/swlh/return-early-pattern-3d18a41bba8).
- Also, by having 3 negative cases it's harder to read (one of the principles [here](https://www.amazon.com/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882)). I would recommend refactoring the code so it reads positive cases first.

For example, maybe something like this:

```
const sanitizeExtraParameters = (name, params) => {
  if (!params) return params;

  for (const key in params) {
    const value = params[key];
      
    // `null` and `undefined` are valid values (e.g. for clearing previously-set values)
    if (value == null || value == undefined) continue;

    if (typeof value !== 'string') {
      console.warn("AppLovinMAXAdView only support string values: " + value + ", deleting value for key: " + key);
      delete params[key];
    }
  }

  return params;
};
```

Reply:
I realized that checking the argument type in the platform requires more changes, especially on Android, so it is done in the JavaScript layer.

Also, I don't know how you want to refactor this.   Please suggest.

```
const setBannerExtraParameter = (adUnitId, key, value) => {
  if ((value !== null) && (value !== undefined) && (typeof value !== 'string')) {
    console.warn("setBannerExtraParameter() supports only string values: " + value);
    return;
  }

  AppLovinMAX.setBannerExtraParameter(adUnitId, key, value);
}
```